### PR TITLE
EE-7205: General HTTP improvements

### PR DESCRIPTION
**Pull request by JuLuzzi**

### CODE REVIEW

File: modules/http-api/src/main/java/org/mule/runtime/http/api/domain/AbstractCaseInsensitiveMultiMap.java [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+    }
+
+    @Override
+    public int hashCode() {
+      return m.hashCode();
+    }
+
+    @Override
+    public Map<String, ? extends List<String>> toListValuesMap() {
+      return m.toListValuesMap();
```
Comment (elrodro83) [inline modules/http-api/src/main/java/org/mule/runtime/http/api/domain/AbstractCaseInsensitiveMultiMap.java]: should this be wrapped as well?

**Comment by JuLuzzi** [comment]
--validate

**Comment by JuLuzzi** [comment]
--validate

**Comment by JuLuzzi** [comment]
--validate

File: modules/http-api/src/main/java/org/mule/runtime/http/api/domain/CaseInsensitiveMultiMap.java [hunk 2]
```diff
@@ -107,4 +121,18 @@ public CaseInsensitiveMultiMap toImmutableMultiMap() {
   public DataType getDataType() {
     return dataType;
   }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) return true;
```
Comment (elrodro83) [inline modules/http-api/src/main/java/org/mule/runtime/http/api/domain/CaseInsensitiveMultiMap.java]: use braces even for one line blocks

File: modules/http-api/src/main/java/org/mule/runtime/http/api/domain/CaseInsensitiveMultiMap.java [hunk 3]
```diff
@@ -124,9 +124,13 @@ public DataType getDataType() {
 
   @Override
   public boolean equals(Object o) {
-    if (this == o) return true;
-    if (!(o instanceof CaseInsensitiveMultiMap)) return false;
-    if (!super.equals(o)) return false;
+    if (this == o) {
+      return true;
+    }
+    if (!(o instanceof CaseInsensitiveMultiMap))
```
Comment (elrodro83) [inline modules/http-api/src/main/java/org/mule/runtime/http/api/domain/CaseInsensitiveMultiMap.java]: here

File: modules/http-api/src/main/java/org/mule/runtime/http/api/domain/CaseInsensitiveMultiMap.java [hunk 4]
```diff
@@ -124,9 +124,13 @@ public DataType getDataType() {
...
+      return false;
+    if (!super.equals(o))
```
Comment (elrodro83) [inline modules/http-api/src/main/java/org/mule/runtime/http/api/domain/CaseInsensitiveMultiMap.java]: and here :)

**Pull request merged**